bin/github_changelog_generator

```diff
@@ -1,4 +1,301 @@
-#! /usr/bin/env ruby
+#!/usr/bin/env ruby
 
-require_relative "../lib/github_changelog_generator"
-GitHubChangelogGenerator::ChangelogGenerator.new.run
+require 'faraday/http_cache'
+require 'sqlite3'
+require 'yaml'
+require 'octokit'
+#require 'shell-spinner'
+
+require_relative "../lib/github_changelog_generator/parser"
+
+module GitHubChangelogGenerator
+  class SQLiteStore
+    @@db = nil
+    @@options = GitHubChangelogGenerator::Parser.options
+    def self.db
+      if @@db.nil?
+        @@db = SQLite3::Database.new(@@options.cache)
+        @@db.execute("CREATE TABLE IF NOT EXISTS httpcache (k PRIMARY KEY, v)")
+        @@db.execute('PRAGMA synchronous = OFF')
+        @@db.execute('PRAGMA journal_mode = MEMORY')
+        @@db.results_as_hash = true
+      end
+
+      return@@db
+    end
+
+    def read(key)
+      v = @@db.get_first_value("SELECT v FROM httpcache WHERE k = ?", key)
+      v = YAML::load(v) if v
+      print(v ? '+' : '-') if @@options.verbose
+      return v
+    end
+
+    def delete(key)
+      @@db.execute("DELETE FROM httpcache WHERE k = ?", key)
+    end
+
+    def write(key, value)
+      @@db.execute('INSERT OR REPLACE INTO httpcache (k, v) VALUES (?, ?)', [key, YAML::dump(value)])
+    end
+  end
+end
+
+DB = GitHubChangelogGenerator::SQLiteStore.db
+OPTIONS = GitHubChangelogGenerator::Parser.options
+
+Octokit.configure{|c|
+  c.access_token = ENV['CHANGELOG_GITHUB_TOKEN'] || ENV[OPTIONS.token] || OPTIONS.token
+}
+Octokit.middleware = Faraday::RackBuilder.new {|builder|
+  builder.use Faraday::HttpCache, shared_cache: false, serializer: Marshal, store: GitHubChangelogGenerator::SQLiteStore.new
```

Comment:
Oh, yes, and here you _did_ use a customized middleware, my apologies.

(Reference to my bullet point item in #321.)